exe/Main.hs

```diff
@@ -194,19 +206,31 @@ cradleToSession cradle = do
     newHscEnvEq env
 
 
-loadSession :: FilePath -> IO (FilePath -> Action HscEnvEq)
-loadSession dir = do
+cradleToSession :: Cradle a -> Action HscEnvEq
+cradleToSession cradle = do
+    cmpOpts <- liftIO $ getComponentOptions cradle
+    let opts = componentOptions cmpOpts
+        deps = componentDependencies cmpOpts
+    -- TODO Handle deps that don't exist yet but might be newly created.
+    existingDeps <- liftIO $ filterM doesFileExist deps
```

Comment:
If you use the Shake doesFileExist then I think you'll automatically end up tracking the absence of the files that haven't yet been created, thus solving the issue you mention.

Reply:
Aha, that's convenient. Done. Had to change the import of `System.Directory.Extra` as a qualified import to disambiguate `doesFileExist`, and update several places in this file.